pubsub/rabbitpubsub: flaky tests when using fake rabbit server

Saw this race on Travis:

```
==================
WARNING: DATA RACE
Write at 0x00c0004e7b30 by goroutine 45:
  runtime.mapdelete_faststr()
      /home/travis/.gimme/versions/go1.11.6.linux.amd64/src/runtime/map_faststr.go:281 +0x0
  gocloud.dev/pubsub/rabbitpubsub.(*fakeChannel).QueueDelete()
      /home/travis/gopath/src/gocloud.dev/pubsub/rabbitpubsub/fake_test.go:352 +0x8c
  gocloud.dev/pubsub/rabbitpubsub.(*harness).CreateSubscription.func1()
      /home/travis/gopath/src/gocloud.dev/pubsub/rabbitpubsub/rabbit_test.go:135 +0xab
  gocloud.dev/pubsub/drivertest.makePair.func1()
      /home/travis/gopath/src/gocloud.dev/pubsub/drivertest/drivertest.go:515 +0x9a
  gocloud.dev/pubsub/drivertest.testCancelSendReceive()
      /home/travis/gopath/src/gocloud.dev/pubsub/drivertest/drivertest.go:393 +0x355
  gocloud.dev/pubsub/drivertest.RunConformanceTests.func1()
      /home/travis/gopath/src/gocloud.dev/pubsub/drivertest/drivertest.go:155 +0x90
  testing.tRunner()
      /home/travis/.gimme/versions/go1.11.6.linux.amd64/src/testing/testing.go:827 +0x162
Previous read at 0x00c0004e7b30 by goroutine 51:
  runtime.mapaccess2_faststr()
      /home/travis/.gimme/versions/go1.11.6.linux.amd64/src/runtime/map_faststr.go:101 +0x0
  gocloud.dev/pubsub/rabbitpubsub.(*fakeChannel).Consume()
      /home/travis/gopath/src/gocloud.dev/pubsub/rabbitpubsub/fake_test.go:221 +0x14f
  gocloud.dev/pubsub/rabbitpubsub.(*subscription).establishChannel.func1()
      /home/travis/gopath/src/gocloud.dev/pubsub/rabbitpubsub/rabbit.go:542 +0x1d2
  gocloud.dev/pubsub/rabbitpubsub.runWithContext.func1()
      /home/travis/gopath/src/gocloud.dev/pubsub/rabbitpubsub/rabbit.go:210 +0x4c
Goroutine 45 (running) created at:
  testing.(*T).Run()
      /home/travis/.gimme/versions/go1.11.6.linux.amd64/src/testing/testing.go:878 +0x659
  gocloud.dev/pubsub/drivertest.RunConformanceTests()
      /home/travis/gopath/src/gocloud.dev/pubsub/drivertest/drivertest.go:155 +0x4a9
  gocloud.dev/pubsub/rabbitpubsub.TestConformance()
      /home/travis/gopath/src/gocloud.dev/pubsub/rabbitpubsub/rabbit_test.go:70 +0x110
  testing.tRunner()
      /home/travis/.gimme/versions/go1.11.6.linux.amd64/src/testing/testing.go:827 +0x162
Goroutine 51 (finished) created at:
  gocloud.dev/pubsub/rabbitpubsub.runWithContext()
      /home/travis/gopath/src/gocloud.dev/pubsub/rabbitpubsub/rabbit.go:210 +0x9f
  gocloud.dev/pubsub/rabbitpubsub.(*subscription).establishChannel()
      /home/travis/gopath/src/gocloud.dev/pubsub/rabbitpubsub/rabbit.go:534 +0x19d
  gocloud.dev/pubsub/rabbitpubsub.(*subscription).ReceiveBatch()
      /home/travis/gopath/src/gocloud.dev/pubsub/rabbitpubsub/rabbit.go:558 +0xf0
  gocloud.dev/pubsub.(*Subscription).getNextBatch.func1()
      /home/travis/gopath/src/gocloud.dev/pubsub/pubsub.go:544 +0x1b9
  gocloud.dev/internal/retry.call()
      /home/travis/gopath/src/gocloud.dev/internal/retry/retry.go:51 +0x1b0
  gocloud.dev/internal/retry.Call()
      /home/travis/gopath/src/gocloud.dev/internal/retry/retry.go:40 +0xbd
  gocloud.dev/pubsub.(*Subscription).getNextBatch()
      /home/travis/gopath/src/gocloud.dev/pubsub/pubsub.go:540 +0x171
  gocloud.dev/pubsub.(*Subscription).Receive.func2()
      /home/travis/gopath/src/gocloud.dev/pubsub/pubsub.go:493 +0xd8
==================
--- FAIL: TestConformance (0.65s)
    <autogenerated>:1: using the fake because the RabbitMQ server is not up (dial error: dial tcp 127.0.0.1:5672: connect: connection refused)
    --- FAIL: TestConformance/TestCancelSendReceive (0.00s)
        testing.go:771: race detected during execution of test
    testing.go:771: race detected during execution of test
FAIL
```

--- a/pubsub/rabbitpubsub/fake_test.go
+++ b/pubsub/rabbitpubsub/fake_test.go
@@ -344,11 +344,17 @@ func (ch *fakeChannel) Close() error {
 }
 
 func (ch *fakeChannel) ExchangeDelete(name string) error {
+	ch.conn.mu.Lock()
+	defer ch.conn.mu.Unlock()
+
 	delete(ch.conn.exchanges, name)
 	return nil
 }
 
 func (ch *fakeChannel) QueueDelete(name string) error {
+	ch.conn.mu.Lock()
+	defer ch.conn.mu.Unlock()
+
 	delete(ch.conn.queues, name)
 	return nil
 }

